src/libraries/System.Reflection.Emit/tests/PersistedAssemblyBuilder/AssemblySaveTypeBuilderTests.cs

```diff
@@ -789,6 +791,143 @@ public void CreateGenericTypeFromMetadataLoadContextSignatureTypes()
+            Assert.Equal(2, paramTypes3.Length);
+            Assert.Equal(typeof(string).FullName, paramTypes3[0].FullName);
+            Assert.Equal(typeof(int).MakeByRefType().FullName, paramTypes3[1].FullName);
+            Assert.Contains(paramTypes3[1].GetRequiredCustomModifiers(), t => t.FullName == typeof(InAttribute).FullName);
+            Assert.Equal(typeof(void).FullName, field3Type.GetFunctionPointerReturnType().FullName);
+            Type[] callingConventions3 = field3Type.GetFunctionPointerCallingConventions();
+            Assert.Contains(callingConventions3, t => t.FullName == typeof(CallConvStdcall).FullName);
+        }
+
+        [Fact]
+        public void ConsumeFunctionPointerFields()
+        {
+            // public unsafe class Container
+            // {
+            //     public static delegate*<int, int, int> Method;
+            // 
+            //     public static int Add(int a, int b) => a + b;
+            //     public static void Init() => Method = &Add;
+            // }
+
+            TempFile assembly1Path = TempFile.Create();
+            PersistedAssemblyBuilder assembly1 = new(new AssemblyName("Assembly1"), typeof(object).Assembly);
+            ModuleBuilder mod1 = assembly1.DefineDynamicModule("Module1");
+            TypeBuilder containerType = mod1.DefineType("Container", TypeAttributes.Public | TypeAttributes.Class);
+            FieldBuilder methodField = containerType.DefineField("Method", typeof(delegate*<int, int, int>), FieldAttributes.Public | FieldAttributes.Static);
+            MethodBuilder addMethod = containerType.DefineMethod("Add", MethodAttributes.Public | MethodAttributes.Static);
+            addMethod.SetParameters(typeof(int), typeof(int));
+            addMethod.SetReturnType(typeof(int));
+            ILGenerator addMethodIL = addMethod.GetILGenerator();
+            addMethodIL.Emit(OpCodes.Ldarg_0);
+            addMethodIL.Emit(OpCodes.Ldarg_1);
+            addMethodIL.Emit(OpCodes.Add);
+            addMethodIL.Emit(OpCodes.Ret);
+            MethodBuilder initMethod = containerType.DefineMethod("Init", MethodAttributes.Public | MethodAttributes.Static);
+            initMethod.SetReturnType(typeof(void));
+            ILGenerator initMethodIL = initMethod.GetILGenerator();
+            initMethodIL.Emit(OpCodes.Ldftn, addMethod);
+            initMethodIL.Emit(OpCodes.Stsfld, methodField);
+            initMethodIL.Emit(OpCodes.Ret);
+            containerType.CreateType();
+            assembly1.Save(assembly1Path.Path);
+
+            // class Program
+            // {
+            //     public static int Main()
+            //     {
+            //         Container.Init();
+            //         return Container.Method(2, 3);
+            //     }
+            // }
+
+            TempFile assembly2Path = TempFile.Create();
+            Assembly assembly1FromDisk = Assembly.LoadFile(assembly1Path.Path);
```

Comment:
This should use `TestAssemblyLoadContext`, similar to https://github.com/dotnet/runtime/blob/8f0e8abd5c5a315de2fb7e8327f1c61d5b8f3dbd/src/libraries/System.Reflection.Emit/tests/PersistedAssemblyBuilder/AssemblySaveILGeneratorTests.cs#L2962-L2964